packages/discovery-react-components/src/components/SearchFacets/utils/buildAggregationQuery.ts

```diff
@@ -1,12 +1,15 @@
-import DiscoveryV2 from 'ibm-watson/discovery/v2';
+import { QueryAggregationWithName } from './searchFacetInterfaces';
 
-export const buildAggregationQuery = (
-  configuration: DiscoveryV2.QueryTermAggregation[]
-): string => {
-  const aggregation = configuration.map(term => {
-    return term.count
-      ? 'term(' + term.field + ',count:' + term.count + ')'
-      : 'term(' + term.field + ')';
+export const buildAggregationQuery = (configuration: QueryAggregationWithName[]): string => {
+  const aggregation = configuration!.map(term => {
+    const termCount = term.count ? ',count:' + term.count : '';
+    const termName = term.name ? ',name:' + term.name : '';
+    let nestedTypeTermAgg = '';
+    if (term.field!.includes('enriched_') && term.field!.includes('entities.text')) {
+      const topLevelTermEntityField = term.field!.split('.')[0];
+      nestedTypeTermAgg = `.term(${topLevelTermEntityField}.entities.type,count:1)`;
```

Comment:
having `count:1` makes me ask -> what happens if we do have more than 1 type for a category? for example if `Apple` is the entity key, couldn't there be two types, `Company` and `Food`? in that case, do we want to drop types by limiting it to `1`?

Reply:
I am limiting to 1 here so that the facet value only shows up underneath 1 category within the whole facet. I'm not sure about the ux of having a facet value show up twice to be selected within the same facet. Would it be selectable beneath each category grouping separately? Would its selection in one category select it in another?